orbit/pkg/table/user_exec/constraints.go

```diff
@@ -0,0 +1,122 @@
+package user_exec
+
+import (
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+type constraintOptions struct {
+	allowedCharacters string
+	allowedValues     []string
+	defaults          []string
+	logger            log.Logger
+}
+
+type GetConstraintOpts func(*constraintOptions)
+
+// WithLogger sets the logger to use
+func WithLogger(logger log.Logger) GetConstraintOpts {
+	return func(co *constraintOptions) {
+		co.logger = logger
+	}
+}
+
+// WithDefaults sets the defaults to use if no constraints were
+// specified. Note that this does not apply if there were constraints,
+// which were invalidated.
+func WithDefaults(defaults ...string) GetConstraintOpts {
+	return func(co *constraintOptions) {
+		co.defaults = append(co.defaults, defaults...)
+	}
+}
+
+func WithAllowedCharacters(allowed string) GetConstraintOpts {
+	return func(co *constraintOptions) {
+		co.allowedCharacters = allowed
+	}
+}
+
+func WithAllowedValues(allowed []string) GetConstraintOpts {
+	return func(co *constraintOptions) {
+		co.allowedValues = allowed
+	}
+}
+
+// GetConstraints returns a []string of the constraint expressions on
+// a column. It's meant for the common, simple, usecase of iterating over them.
+func GetConstraints(queryContext table.QueryContext, columnName string, opts ...GetConstraintOpts) []string {
```

Comment:
These types and methods should go to a separate package `query_constraints` or some better name of course :)

Reply:
Seems you intended it that way?

`user_exec.go` is using a package not present on this PR `github.com/kolide/launcher/pkg/osquery/tables/tablehelpers`?